@here/harp-mapview/lib/MapView.ts

```diff
@@ -1154,7 +1155,9 @@ export class MapView extends EventDispatcher {
         // will initialize with an empty theme and updated when theme is loaded and set
         this.m_textElementsRenderer = this.createTextRenderer();
 
-        this.setTheme(getOptionValue(this.m_options.theme, MapViewDefaults.theme));
+        this.m_promises.push(
```

Comment:
I am a bit confused about the approach, where I see that having a promise to wait for is a nice alternative to attaching to the THEME_LOADED Event, I dont see why we need to add the addDatasources/promises array as addDataSource is already async and returns a Promise that can be kept

Reply:
Yes, I originally exposed the promise from `addDataSource`  in `MapViewEnvironment`, but @atomicsulfate didn't like that approach.